### ISSUE TITLE
Return a wrong number of frames FileLoop

### ISSUE THREAD

**Issue by QuangTung97** [thestk/stk#78]
I tried to get number of frames by using FileLoop but getSize() return a wrong number. 
But when I tried with FileWvIn, I got a correct result.

**Comment by garyscavone** [comment]
What value did FileLoop provide (vs. FileWvIn)?

**Comment by radarsat1** [comment]
I notice that getSize is definitely different between FileLoop and FileWvIn.

The function FileLoop::getSize() returns `data_.size()`, which is the size of its data buffer, not the size of the file it is loading, which is not the same as what the comment says.

```
  //! Return the file size in sample frames.
  unsigned long getSize( void ) const { return data_.frames(); };
```

Also, what is the size of the file?  Because the FileLoop object has a maximum size of data that it will load at once, called the chunkThreshold (argument to the constructor), and the value returned may be different based on this.

The size of the buffer is determined by,

```
  // Determine whether chunking or not.
  if ( file_.fileSize() > chunkThreshold_ ) {
    chunking_ = true;
    chunkPointer_ = 0;
    data_.resize( chunkSize_ + 1, file_.channels() );
  }
  else {
    chunking_ = false;
    data_.resize( file_.fileSize() + 1, file_.channels() );
  }
```

@garyscavone do you know where the `+ 1` comes from?

I think you can get the file size if you call the base class,

```
FileLoop *loop;
loop = .. ;
int size = loop->FileWvIn::getSize();
```

Maybe a solution though could be to not override `getSize()` in FileLoop and instead add a function, `getChunkSize`.

**Comment by garyscavone** [comment]
I agree that the getSize() function should not be overridden in FileLoop. And I’m not sure that there even needs to be a function that returns the chunkSize. There is a chance that we may break something else in STK by changing it but it could probably be easily fixed.

The “+1” in sizing the data buffer comes from the fact that it is easier to interpolate the data when looping by putting an extra sample frame at the end of the buffer that is a copy of the first value in the buffer. 

I suggest we simply delete the implementation of getSize() in FileLoop.

**Comment by garyscavone** [comment]
Fixed in commit 6ce3ac9e10fde4adbb863f4fa04a6d0e185b242c
